libs/d3d12/main.c

```diff
@@ -46,302 +48,49 @@
 static pthread_once_t library_once = PTHREAD_ONCE_INIT;
 
 #ifdef _WIN32
-static HMODULE vulkan_module = NULL;
+static HMODULE d3d12core_module = NULL;
 #else
-static void *vulkan_module = NULL;
+static void *d3d12core_module = NULL;
 #endif
-static PFN_vkGetInstanceProcAddr vulkan_vkGetInstanceProcAddr = NULL;
 
-static void load_vulkan_once(void)
+static IVKD3DCoreInterface* core = NULL;
+
+static void load_d3d12core_once(void)
 {
-    if (!vulkan_module)
+    if (!d3d12core_module)
     {
-#ifdef _WIN32
-        /* If possible, load winevulkan directly in order to bypass
-         * issues with third-party overlays hooking the Vulkan loader */
-        static const char *vulkan_dllnames[] =
-        {
-            "winevulkan.dll",
-            "vulkan-1.dll",
-        };
-
-        unsigned int i;
-
-        for (i = 0; i < ARRAY_SIZE(vulkan_dllnames); i++)
-        {
-            vulkan_module = LoadLibraryA(vulkan_dllnames[i]);
+        PFN_D3D12_GET_INTERFACE d3d12core_D3D12GetInterface = NULL;
 
-            if (vulkan_module)
-            {
-                vulkan_vkGetInstanceProcAddr = (void*)GetProcAddress(vulkan_module, "vkGetInstanceProcAddr");
-
-                if (vulkan_vkGetInstanceProcAddr)
-                    break;
-
-                FreeLibrary(vulkan_module);
-                vulkan_module = NULL;
-            }
-        }
+#ifdef _WIN32
+        if (d3d12core_module = LoadLibraryA(SONAME_D3D12CORE))
+            d3d12core_D3D12GetInterface = (PFN_D3D12_GET_INTERFACE)GetProcAddress(d3d12core_module, "D3D12GetInterface");
 #else
-        vulkan_module = dlopen(SONAME_LIBVULKAN, RTLD_LAZY);
-        vulkan_vkGetInstanceProcAddr = (PFN_vkGetInstanceProcAddr)dlsym(vulkan_module, "vkGetInstanceProcAddr");
+        if (d3d12core_module = dlopen(SONAME_D3D12CORE, RTLD_LAZY))
+            d3d12core_D3D12GetInterface = (PFN_vkGetInstanceProcAddr)dlsym(vulkan_module, "D3D12GetInterface");
```

Comment:
This is not ideal on Linux since we probably have to mess around with RPATH to make sure d3d12core is found without too much hassle with LD_LIBRARY_PATH and stuff. If we have to have the split on native build, d3d12 should link directly to d3d12core with RPATH to avoid any issues.

Reply:
Sorry what is RPATH?

Reply:
When loading a library on Linux, it looks in LD_LIBRARY_PATH, but that will not include the directly of the file itself. The usual way to work around that is to use RPATH to add more implicit lookup folders when linking, but I don't think that applies to dlopen ... See https://en.wikipedia.org/wiki/Rpath.

Reply:
I have resolved this by making d3d12 explicitly link against d3d12core on Linux which automatically handles any cases that would normally need LD_LIBRARY_PATH to solve.

Still need to dlopen it and getproc as D3D12 + D3D12Core both export D3D12GetInterface, so linker would get confused otherwise.